multi-agent/lib/mas/elements/providers/mcp_server_client/mcp_server_client.py

```
@@ -202,6 +202,22 @@ async def call_tool(
         """
         return await self.tools.call_tool(tool_name, arguments)
 
+    async def list_resources(self):
+        """List all resources available on the MCP server."""
+        try:
+            session = self.transport._session
+            result = await session.list_resources()
+            return result.resources
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)